packages/block-library/src/navigation/build-navigation-label.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+
+function buildNavigationLabel( title, id, status ) {
+	if ( ! title ) {
+		/* translators: %s is the index of the menu in the list of menus. */
+		return sprintf( __( '(no title %s)' ), id );
+	}
+
+	if ( status === 'publish' ) {
+		return decodeEntities( title );
```

Comment:
Nit: can we do `decodeEntities( title )` once before the `if ( status === 'publish' ) {`. That way we don't have any chance of refactoring loosing the decoding.